### ISSUE TITLE
Standardise architecture compatibility reports

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockon-registry#309]
We have yet to establish a proper arch compatibility element within our Rock-on config but we will, never-the-less, need a full run down on the current known compatibility. It is proposed that we, in the mean time, add to all Rock-ons their respective known architecture compatibility and standardise on:

"amd64" and "arm64" as we have many compatible but confusing references.

These too seem appropriate as they are the ones used in dockerhub. I personally like x86_64 but it does't feature there and so is likely just irrelevant now.

I also propose that we maintain this order of appearance and cut the wording to an absolute minimum for now as later these entries will have their own UI elements and we may even be able to strip them out of the description to facilitate our transition and maintain compatibility with the older (read current) format once we have a new "arch" Rock'on element.

We should also remove all other references such as other arm references as these are not supported in Rockstor anyway and so are irrelevant noise. An example of these 'other' entries are:
"arm/v7 and arm/v6"
"arm32v7 and arm64v8"
"arm32v7 and arm64v8"
"armhf"

**Comment by Hooverdan96** [comment]
@phillxnet, @FroggyFlox my suggestion is to have two flavors of some formulation like this, that would always be at the end of the description of a Rockon:

<p>Based on the official docker image: <a href='https://hub.docker.com/r/kanboard/kanboard' target='_blank'>https://hub.docker.com/r/kanboard/kanboard</a>, available on amd64 and arm64 architecture.",

<p>Based on a custom docker image: <a href='https://hub.docker.com/r/put_image_name' target='_blank'>https://hub.docker.com/r/put_image_name</a>, available on amd64 and arm64 architecture.",

During the pull request/discussion process the community will have already landed on why/whether custom image is to be used or not.

The above is just a suggestion to formalize a bit more (since most Rockons have to be touched anyway for the architecture cleanup/clarification.

It there a better way to highlight architecture denomination, so it can be more easily split in the future, if we decide to have a separate json tag, so the Rockons can be better presented in a tabular fashion?

**Comment by Hooverdan96** [comment]
Hi again, I have started with updating some of the Rockon jsons. Do you prefer individual Pull Requests for each of them, or does it make sense for these cosmetic changes to keep them in one Pull Request (so I would keep adding to the #310 request, or delete this one and submit it all in one at the end)?

**Comment by phillxnet** [comment]
@Hooverdan96 Thanks for picking this one up.
Re:
> Do you prefer individual Pull Requests for each of them, or does it make sense for these cosmetic changes to keep them in one Pull Request
I say we do as many as you fancy tending to all in one pull request myself. But @FroggyFlox is more master of Rock-on proceedings so we could see what he thinks. As you say it's just a cosmetic, but it does almost all, if not all Rock-ons. I can see how a system deals with this when we get to that however. Should be OK as we update the db on all new Rock-ons on first setting up the Rock-ons system but they are all fresh db records and this would be all existing ones being updated. But we can see.

**Comment by Hooverdan96** [comment]
sounds good, I'll wait for @FroggyFlox's input if he has time to think about it. Otherwise, I'll start with as many as I feel like it. And worst case, we'll split them up in groups or individually (not being a Git expert really, but I assume stuff like that can be done).

**Comment by phillxnet** [comment]
@Hooverdan96 
> And worst case, we'll split them up in groups or individually (not being a Git expert really, but I assume stuff like that can be done).

Steady, we don't want anything too git complicated. You could as you seem to be doing, do a commit for each Rock-on. That may help just with managing your own progress really. But you can also squash commits (together) later-on, but that would likely require a force push to your GitHub origin. Or deleting the GitHub origin and re-pushing, which can end up closing the pr I think. You can experiment to see as you can always open a new pr. I generally like to avoid force pushes and have often done the squash-delete-origin thing but mostly before starting a pull request. But we are not that fussy regarding git style etc in the Rock-ons, docs, and likely the website repos as that can present an unreasonable barrier to folks.

Git is great for monitoring/tracking your own progress, as you know, so I say don't let our repo preference interfere with that. Especially given we have the delete origin squash say all working commits, and re-push option, all-be-it with the pr having to be recreated. But at least then you have the backup option of pushing regularly to the GitHub 'cloud', and ability to chunk your own work while it's in-progress. Plus the freedom to drop in silly little commits of stuff forgotten (or typos), that can later be squashed just prior to submitting the real pr so it ends up looking tidy. Again I like the JetBrains PyCharm git management capabilities here. And I hear they are quite well respected generally.

I'm no git expert myself and definitely have to tread carefully especially with the new master/testing core branch split we had to do recently. I've also avoided any additional complexity like git sub-modules, commonly used in Hugo for instance, so that a theme is held separate from the website repo it is used in. All super capable but too complicated.

So I say, given you have currently many identically named commits, that you do whatever works for you, then have a look at squashing everything once you have it all sorted at a stage you are happy with. As this issue could all be addressed in a single commit with the name of the issue. And when I have ended up throwing in other possibly unrelated stuff I pop in a commit note to this end. But ultimately do as you fancy as it's great to have your assistance here, as always.
